audbcards/core/dataset.py

```diff
@@ -64,6 +94,33 @@ def __init__(
                 audeer.path(self.cache_root, name, other_version)
             )
 
+        # save instance to cache
+        object_cache = audeer.path(
+            self.cache_root,
+            'dataset',
+            name,
+            version,
+            'dataset.pkl',
+        )
+        self._save_pickled(object_cache)
+
+    @staticmethod
+    def _load_pickled(path: str):
+        r"""Load pickled object instance."""
+        if not os.path.exists(path):
+            raise FileNotFoundError()
+
+        with open(path, "rb") as f:
+            return pickle.load(f)
+
+    def _save_pickled(self, path: str):
+        """Save object instance to path as pickle."""
+        audeer.mkdir(os.path.dirname(path))
```

Comment:
You execute the same in the `datasetcache` decorator. I guess it is only needed at one place.

Reply:
The decorator is gone altogeher.